src/Microsoft.TestPlatform.Extensions.BlameDataCollector/NetClientHangDumper.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.TestPlatform.Extensions.BlameDataCollector
+{
+    using System;
+    using System.Diagnostics;
+    using System.IO;
+    using System.Linq;
+    using System.Runtime.InteropServices;
+    using Microsoft.Diagnostics.NETCore.Client;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+    using Microsoft.VisualStudio.TestPlatform.Utilities;
+
+    internal class NetClientHangDumper : IHangDumper
+    {
+        public void Dump(int processId, string outputDirectory, DumpTypeOption type)
+        {
+            var process = Process.GetProcessById(processId);
+            var processTree = process.GetProcessTree();
+
+            if (EqtTrace.IsVerboseEnabled)
+            {
+                if (processTree.Count > 1)
+                {
+                    EqtTrace.Verbose("NetClientHangDumper.Dump: Dumping this process tree (from bottom):");
+                    ConsoleOutput.Instance.Information(false, "Blame: Dumping this process tree (from bottom):");
+
+                    foreach (var p in processTree.OrderBy(t => t.Level))
```

Comment:
Does that ordering happen in each iteration? Would be better to do it before the loop.

Reply:
It does not, it orders once, and foreach iterates over the resulting IEnumerable, if it did not you'd get collection modified exception.